.github/workflows/ci.yml

```diff
@@ -0,0 +1,67 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+    tags:
+      - "*.*.*"
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.ivy2
+            ~/.sbt
+      - name: Build
+        run: sbt package
+      - uses: actions/upload-artifact@v2
+        with:
+          name: jars
+          path: '*/target/*/*.jar'
+          retention-days: 1
+  publish:
+    if: github.event_name != 'pull_request'
+    needs: build-and-test
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.ivy2
+            ~/.sbt
+      - name: Set SBT_OPTS
+        run: bash gen_sbt_opts.sh
+      - name: Publish to Sonatype
+        run: bash publish.sh
+        env:
+          PGP_SECRET: ${{ secrets.PGP_SECRET }}
```

Comment:
have these secrets been added to the repo?

Reply:
no. they have not been added. but I wasn't sure how to get them. is this setup in travis somewhere?